security/advancedtls/advancedtls.go

```diff
@@ -215,9 +233,7 @@ func (o *ServerOptions) config() (*tls.Config, error) {
 		ClientAuth:     clientAuth,
 		Certificates:   o.Certificates,
 		GetCertificate: o.GetCertificate,
-	}
-	if o.RootCACerts != nil {
-		config.ClientCAs = o.RootCACerts
+		ClientCAs:      clientCAs,
```

Comment:
I think it kind of makes sense to set `clientCAs` only when `o.RootCACerts != nil`, since there is no point to put a nil field in `tls.Config`.  Setting nil field in `tls.Config` doesn't mean the end of the world, because we have creds reloading in the hope of replacing that nil field, but I think it's clearer if we don't set it in the first place.
Client side might also need similar settings. Can you also help update that?